omegaconf/omegaconf.py

```diff
@@ -101,25 +101,55 @@ def SI(interpolation: str) -> Any:
 
 
 def register_default_resolvers() -> None:
-    def env(key: str, default: Any = _EMPTY_MARKER_) -> Any:
+    # # DEPRECATED: remove in 2.2
+    def legacy_env(key: str, default: Optional[str] = None) -> Any:
+        warnings.warn(
+            "The `env` resolver is deprecated, see https://github.com/omry/omegaconf/issues/573",
+        )
+
+        try:
+            return decode_primitive(os.environ[key])
+        except KeyError:
+            if default is not None:
+                # We handle the "null" string in a special way so as to preserve
+                # the old behavior of `env` while keeping the fix from #230.
+                if default.lower() == "null":
+                    return None
+                else:
+                    return decode_primitive(default)
+            else:
+                raise ValidationError(f"Environment variable '{key}' not found")
+
+    def env(key: str, default: Any = _EMPTY_MARKER_) -> Optional[str]:
         try:
-            val_str = os.environ[key]
+            return os.environ[key]
         except KeyError:
             if default is not _EMPTY_MARKER_:
+                if default is not None and not isinstance(default, str):
+                    raise ValidationError(
+                        f"The default value of the `oc.env` resolver must be a string or "
+                        f"None, but `{default}` is of type {type(default).__name__}"
+                    )
                 return default
             else:
                 raise ValidationError(f"Environment variable '{key}' not found")
 
-        # We obtained a string from the environment variable: we try to parse it
-        # using the grammar (as if it was a resolver argument), so that expressions
-        # like numbers, booleans, lists and dictionaries can be properly evaluated.
-        try:
-            parse_tree = parse(
-                val_str, parser_rule="singleElement", lexer_mode="VALUE_MODE"
+    def decode(expr: Optional[str]) -> Any:
```

Comment:
I think it's time to create a module for built in resolvers.
Maybe:
```
omegaconf/built_in_resolvers.py
```

Also, to make the built in resolvers first class functions in that module and not hidden in a function.

Reply:
Yeah, I already have that in a follow-up PR (WIP), so going to resolve it for now